### PR TITLE
DOC: Improve docstring for numpy.linalg.lstsq

### PR DESCRIPTION
**Pull request by pdebuyl**
Address the possibly misleading docstring, see discussion in #15874

Start the docstring with """Computes the vector x that approximatively
solves the equation :math:`a x = b`.""" instead of """Solves the
equation :math:`a x = b`""".

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
Only one small comment about `:math:` inline vs. inline literal syntax - just an idea, feel free to ignore.

One thing I noticed when comparing to the [corresponding scipy docstring](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.lstsq.html): `scipy` claims to minimize the L2 norm while the numpy version claims to minimize the *square* of the L2 norm. I'm not a linear algebra expert, but it seems like `numpy.linalg.lstsq` relies on the `?gelsd` functions, whose docs state that they minimize the L2-norm. In practice I don't know if there's a difference, but it might be worth double-checking that the docs accurately reflect what's being computed under the hood.

**Comment by pdebuyl** [comment]
@rossbar I saw other uses of "@" in eig for instance, why not.

Regarding the norm vs square norm, lapack says that it minimizes the norm, see http://www.netlib.org/lapack/lug/node27.html so I am modifying it as you suggest.

File: numpy/linalg/linalg.py
Comment (rossbar) [inline numpy/linalg/linalg.py:2181]: I actually liked the `:math:` rendering here better since this is a *slightly* more complicated expression than `a @ x = b`. I think it's fine either way though!

**Review by rossbar — APPROVED** [review]
LGTM - a minor personal preference for `math` rendering of the L2-norm expression, but that can be safely ignored.

**Comment by pdebuyl** [comment]
I had no strong opinion on that, reversed!

**Comment by pdebuyl** [comment]
With the square removed again in the formula, sorry for the noise.

**Comment by pdebuyl** [comment]
@rossbar the PR is already approved but since there have been modifications, I guess that @seberg will wait for an additional approval to merge.

**Comment by seberg** [comment]
Thanks everyone, @pdebuyl and Ross for the review! If someone finds some more to bikeshed/improve we can make a new PR. It seems the only way to clarify things more would be to expand a fair bit in the `Notes` sections, and with links to wikipedia or so... which is a different project.

**Pull request merged**
